toolchain/lexer/numeric_literal.h

```diff
@@ -12,7 +12,7 @@
 #include "llvm/ADT/StringRef.h"
 #include "toolchain/diagnostics/diagnostic_emitter.h"
 
-namespace Carbon {
+namespace Carbon::Lex {
 
 // A numeric literal token that has been extracted from a source buffer.
 class LexedNumericLiteral {
```

Comment:
Should we drop the `Lexed` prefix now that the namespace effectively says the same thing?